### ISSUE TITLE
Mill Client-Daemon mode hanging in WSL2 Ubuntu 18.04

### ISSUE THREAD

**Issue by lihaoyi** [com-lihaoyi/mill#874]
As far as I can tell so far, this is because the following code that tries to get a file lock fails silently on WSL2 Ubuntu:

```scala
val raf = new java.io.RandomAccessFile("/home/ubuntu/hello", "rw")
val chan = raf.getChannel
val lock = chan.tryLock()
```

I can run this in two separate Ammonite REPLs and both of them will claim to successfully take the lock. In contrast, doing this on OS-X or a proper Ubuntu machine will have the second call to `tryLock()` return `null`.

This ends up confusing the client-server handshake, where the client waits for the server to claim the `processLock` to indicate that it is up and running: since the client always sees the lock as free, it thinks the server never starts and just hangs

Seems to  be caused by this underlying issue: https://github.com/microsoft/WSL/issues/1927

**Comment by sake92** [comment]
Seems to work now, mabye https://github.com/com-lihaoyi/mill/pull/1230 fixed it.. 😄

**Comment by lefou** [comment]
Probably fixed by #2901.
